File: backend/db_test.py
import asyncio
from motor.motor_asyncio import AsyncIOMotorClient
from bson import ObjectId

async def main():
    uri = "mongodb+srv://yasha1b2c:8880Mgyash@cluster0.z1p6i.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
    client = AsyncIOMotorClient(uri)
    db = client.smart_shiksha
    
    test = await db.mock_tests.find_one()
    print(f"Sample test _id: {test['_id']} (type: {type(test['_id'])})")
    
    qs = await db.test_questions.find({"test_id": str(test["_id"])}).to_list(10)
    print(f"Questions found with string test_id: {len(qs)}")
    
    for q in qs[:1]:
        print(f"Sample Question test_id: {q['test_id']} (type: {type(q['test_id'])})")
        print(f"Sample Question options: {q.get('options')} (type: {type(q.get('options'))})")
        
    client.close()

if __name__ == "__main__":
    asyncio.run(main())

File: backend/db_test.js
const { MongoClient, ObjectId } = require("mongodb");

const uri = "mongodb+srv://yasha1b2c:8880Mgyash@cluster0.z1p6i.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
const client = new MongoClient(uri);

async function run() {
  try {
    await client.connect();
    const db = client.db("smart_shiksha");
    const test = await db.collection("mock_tests").findOne({});
    console.log("Sample Test ID:", test._id.toString());
    
    // Check questions by string and by ObjectId
    const questionsString = await db.collection("test_questions").find({ test_id: test._id.toString() }).toArray();
    console.log("Questions found (string ID):", questionsString.length);
    
    // Check what the first question looks like
    const firstQ = await db.collection("test_questions").findOne();
    console.log("Sample question test_id field type:", typeof firstQ?.test_id);
    console.log("Sample question test_id value:", firstQ?.test_id);

  } finally {
    await client.close();
  }
}
run().catch(console.dir);
